README.md

```diff
@@ -151,6 +151,9 @@ kong_nginx_http_current_connections{state="writing"} 1
 # HELP kong_nginx_metric_errors_total Number of nginx-lua-prometheus errors
 # TYPE kong_nginx_metric_errors_total counter
 kong_nginx_metric_errors_total 0
+# HELP kong_upstream_target_health Is upstream target healthy?, 0 is Healthchecks off, 1 is Healthy, 2 is Unhealthy, 3 is DNS Error
+# TYPE kong_upstream_target_health gauge
+kong_upstream_target_health{upstream="<upstream_name>",target="<target_host>:<target_port>"} 1
```

Comment:
This seems very weird from Prometheus perspective.

The state should be a label and not the value of the counter.
Isn't the following more idiomatic in Prometheus?
```
kong_upstream_target_health{upstream="foo.com.internal" target="10.0.1.28" state="healthy"} 1
kong_upstream_target_health{upstream="foo.com.internal" target="10.0.1.29" state="healthy"} 1
kong_upstream_target_health{upstream="foo.com.internal" target="10.0.1.30" state="unhealthy"} 1
```

Reply:
I agree with you, but It would increase the cardinality, each target from an upstream would have at least 3 metrics (state="healthy|unhealthy|dns_error"). Do you think it is OK? I can change the PR here.

Reply:
> but It would increase the cardinality

Good point. How many targets do you usually have in your setup? Let's say 5000, then you get 15000 metrics, and that is well within the range of what Prometheus could handle. Please let me know if I'm missing something here.


The reason I think we should do the change is because users would want to filter and know which targets are healthy vs unhealthy.

Reply:
ping @fffonion @DennoVonDiesel

Reply:
@hbagdi it took me a while to figure it out how to update an upstream in the spec file, but now it is all right. I have written tests for all 3 states.

Reply:
question: do we still have state for `0 is Healthchecks off`?

Reply:
no, we dont. I based on this comment https://github.com/Kong/kong-plugin-prometheus/pull/70/files#r347012571 but I can expose it.

Reply:
To be honest, I prefer expose the state "Healthchecks off" too. It could be an alert for misconfiguration.

Reply:
i'm +1 with exposing the healthchecks off state

Reply:
@fffonion done